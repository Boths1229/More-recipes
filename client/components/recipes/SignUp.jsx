import React, { Component } from 'react';
import { render } from 'react-dom';
import backgroundImage from '../../public/images/blog-img-01.jpg';

/**
 * @class SignUp
 * @description Create user account
 */
class SignUp extends Component {
  /**
   * @description render - renders the class component
   * @return {object} returns an object
   */
  componentWillMount() {
    document.body.style.backgroundImage = `url(${backgroundImage})`;
  }
  /**
   * @description render - renders the class component
   * @return {object} returns an object
   */
  render() {
    return (
      <div className="container">
        <div className="pt-5" style={{ textAlign: 'center' }}>
          <form onSubmit={this.handleSubmit} className="m-x-auto app-login-form card-login">
            <h1 className="signin-text"> More-recipe</h1>
            <section className="signin-text">
              <h3>Create an account</h3>
            </section>
            <section className="pb-4 pt-3">
              <input
                className="form-control"
                placeholder="Username"
                type="text"
                required
              />
            </section>
            <section className="pb-4">
              <input
                className="form-control"
                placeholder="Email"
                type="email"
                required
              />
            </section>
            <section className="pb-4">
              <input
                className="form-control"
                placeholder="Password"
                type="password"
                required
              />
            </section>
            <section className="pb-4">
              <input
                className="form-control"
                placeholder="Confirm password"
                type="password"
                required
              />
            </section>
            <div>
              <p>
                <small>
                  By clicking &#39;
                    Create Account&#39;,
                    you are agreeing to our
                  <a
                    href="Policy"
                    id="text-dec"
                  > Policy
                  </a>
                  <span>  and  </span>
                  <a href="terms" id="text-dec">
                    Terms &amp; Conditions
                  </a>
                </small>
              </p>
            </div>
            <div>
              <button className="btn btn-block btn-orange">
                <a href="signin.html" className="signin-btn" id="signn-btn">Create Account</a>
              </button>
            </div>
            <div style={{ textAlign: 'left' }}>
              <a href="f" id="text-dec">Forgotten your password?</a>
            </div>
            <div>
              <hr />
            </div>
            <footer>
              <a href="signin.html" id="text-dec">Sign In</a>
            </footer>
          </form>
        </div>
      </div>
    );
  }
}

export default SignUp;
