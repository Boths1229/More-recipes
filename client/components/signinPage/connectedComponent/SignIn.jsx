import React, { Component } from 'react';
import PropTypes, { any } from 'prop-types';
import lodash from 'lodash';
import { connect } from 'react-redux';
import Form from '../../signinPage/Form';
import backgroundImage from '../../../public/images/recipe-img-bg.jpeg';
import signInAction from '../../../actionController/signin';

/**
 * @class SignIn
 *
 * @description Log in existing Users
 */
export class SignIn extends Component {
  /**
   * @description create an instance of Signin
   *
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.state = {
      responseMessage: '',
    };
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }
  /**
   * @description mount - mount background image
   *
   * @returns {void} mount an image in the background
   */
  componentWillMount() {
    document.body.style.backgroundImage = `url(${backgroundImage})`;
    if (this.props.isAuthenticated) {
      this.props.history.push('/recipes');
    }
  }

  /**
   * @description mount - mount background image
   *
   * @returns {object} mount an image in the background
   */
  componentDidMount() {
    document.body.style.backgroundImage = `url(${backgroundImage})`;
  }

  /**
   * @description update the state of error message and also redirect to a new page
   *
   * @param {object} nextProps
   *
   * @returns {void}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.isAuthenticated) {
      this.props.history.push('/recipes');
    }
    if (!lodash.isEmpty(nextProps.signinResponse.errorMessage)) {
      this.setState({
        responseMessage: nextProps.signinResponse.errorMessage
      });
    }
  }

  /**
   * @description remove background image before redirecting to a new page
   *
   * @returns {void}
   */
  componentWillUnmount() {
    document.body.style.backgroundImage = '';
  }


  /**
   * @description Handles the input value changes
   *
   * @param {object} event
   *
   * @returns {void} update state
   *
   * @memberof SignIn
   */
  handleChange(event) {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }


  /**
   * @description log a user in when form is submitted
   *
   * @param {object} event
   *
   * @returns {void} calls signInAction
   *;
   * @memberof SignIn
   */
  handleFormSubmit(event) {
    event.preventDefault();
    this.state.responseMessage = '';
    this.props.signInAction(this.state);
  }

  /**
   * @description render - renders the class component
   *
   * @returns {JSX} returns JSX
   */
  render() {
    return (
      <Form
        value={this.state}
        onChange={this.handleChange}
        handleSubmit={this.handleFormSubmit}
      />
    );
  }
}

SignIn.defaultProps = {
  isAuthenticated: false
};

SignIn.propTypes = {
  signInAction: PropTypes.func.isRequired,
  signinResponse: PropTypes.objectOf(any).isRequired,
  history: PropTypes.objectOf(any).isRequired,
  isAuthenticated: PropTypes.bool
};

/**
 * @description make state available to sigIn class as props
 *
 * @param {object} state
 *
 * @returns {void}
 */
const mapStateToProps = state => ({
  signinResponse: state.authReducer,
  isAuthenticated: state.authReducer.isAuthenticated,
});

export default connect(mapStateToProps, { signInAction })(SignIn);
