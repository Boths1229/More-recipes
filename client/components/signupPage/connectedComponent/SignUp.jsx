import React, { Component } from 'react';
import lodash from 'lodash';
import miniToastr from 'mini-toastr';
import { connect } from 'react-redux';
import PropTypes, { any } from 'prop-types';
import signUpAction from '../../../actionController/signup';
import Form from '../../signupPage/Form';
import backgroundImage from '../../../public/images/recipe-img-bg.jpeg';

/**
 * @class SignUp
 *
 * @description Create user account
 */
export class SignUp extends Component {
  /**
   * @description create an instance of Signup
   *
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.state = {
      password: '',
      confirmPassword: '',
      responseMessage: '',
    };
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  /**
   * @description mount background image
   *
   * @returns {undefined} mount a background image
   */
  componentWillMount() {
    document.body.style.backgroundImage = `url(${backgroundImage})`;
  }

  /**
   * @description mount background image
   *
   * @returns {void} mount a background image
   */
  componentDidMount() {
    document.body.style.backgroundImage = `url(${backgroundImage})`;
  }

  /**
   * @description redirect user to recipe page and set the state of error messages
   *
   * @param {object} nextProps
   *
   * @returns {void} set the state of response message
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.isAuthenticated) {
      this.props.history.push('/recipes');
    }
    if (!lodash.isEmpty(nextProps.signupResponse.errorMessage)) {
      this.setState({
        responseMessage: nextProps.signupResponse.errorMessage
      });
    }
  }

  /**
   * @description remove background image before redirecting to a new page
   *
   * @returns {void} remove background Image
   */
  componentWillUnmount() {
    document.body.style.backgroundImage = '';
  }

  /**
   * @description Handles the input value changes
   *
   * @param {object} event
   *
   * @returns {void} call setState
   *
   * @memberof SignUp
   */
  handleChange(event) {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }
  /**
   * @description Sign up a user when form is submitted
   *
   * @param {object} event
   *
   * @returns {void} calls signUpAction
   *
   * @memberof SignUp
   */
  handleFormSubmit(event) {
    event.preventDefault();
    this.state.responseMessage = '';
    if (this.state.password === this.state.confirmPassword) {
      this.props.signUpAction(this.state);
    } else {
      miniToastr.init();
      miniToastr.error('Passwords do not match');
    }
  }
  /**
   * @description render - renders the class component
   *
   * @returns {JSX} returns jsx
   */
  render() {
    return (
      <Form
        value={this.state}
        onChange={this.handleChange}
        handleSubmit={this.handleFormSubmit}
      />
    );
  }
}
SignUp.propTypes = {
  signUpAction: PropTypes.func.isRequired,
  signupResponse: PropTypes.objectOf(any).isRequired,
  history: PropTypes.objectOf(any).isRequired,
  isAuthenticated: PropTypes.bool.isRequired
};

/**
 * @description make state available to signUp class as props
 *
 * @param {object} state
 *
 * @returns {void} mapStateToProps
 */
const mapStateToProps = state => ({
  signupResponse: state.authReducer,
  isAuthenticated: state.authReducer.isAuthenticated
});


export default connect(mapStateToProps, { signUpAction })(SignUp);
