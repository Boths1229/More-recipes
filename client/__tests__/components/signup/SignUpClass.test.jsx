import React from 'react';
import { shallow } from 'enzyme';
import { SignUp } from '../../../components/signupPage/connectedComponent/SignUp';

describe('Signup connected component', () => {
  it('matches snapshot', () => {
    const output = shallow(<SignUp />);
    expect(output).toMatchSnapshot();
  });
  describe('instance and lifecycle methods', () => {
    const props = {
      history: {
        push: jest.fn()
      },
      signUpAction: jest.fn(),
      isAuthenticated: false,
      signupResponse: {}
    };
    it('calls componentWillRecieveProps', () => {
      const wrapper = shallow(<SignUp {...props} />);
      const cwrp = jest.spyOn(wrapper.instance(), 'componentWillReceiveProps');
      wrapper.setProps({ isAuthenticated: true, signupResponse: { errorMessage: 'error message' } });
      expect(cwrp).toHaveBeenCalled();
    });
    it('call component will unmount', () => {
      const wrapper = shallow(<SignUp {...props} />);
      const cwrp = jest.spyOn(wrapper.instance(), 'componentWillUnmount');
      wrapper.unmount();
      expect(cwrp).toHaveBeenCalled();
    });
    it('handleChanges', () => {
      const wrapper = shallow(<SignUp {...props} />);
      const setState = jest.spyOn(wrapper.instance(), 'setState');
      wrapper.instance().handleChange({ target: { name: 'username', value: '' } });
      expect(setState).toHaveBeenCalledWith({ username: '' });
    });
    it('handleSubmit for else condition', () => {
      const wrapper = shallow(<SignUp {...props} />);
      wrapper.setState({ password: 'all', confirmPassword: 'password' });
      wrapper.instance().handleFormSubmit({ preventDefault: jest.fn() });
      expect(props.signUpAction).not.toBeCalled();
    });
    it('handleSubmit', () => {
      const wrapper = shallow(<SignUp {...props} />);
      wrapper.instance().handleFormSubmit({ preventDefault: jest.fn() });
      expect(props.signUpAction).toBeCalled();
    });
  });
});
