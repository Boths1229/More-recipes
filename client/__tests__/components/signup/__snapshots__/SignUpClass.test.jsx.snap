// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup connected component matches snapshot 1`] = `
<Form
  handleSubmit={[Function]}
  onChange={[Function]}
  value={
    Object {
      "confirmPassword": "",
      "password": "",
      "responseMessage": "",
    }
  }
/>
`;
